src/java.base/share/classes/java/lang/reflect/code/op/ExternalOpContent.java

```diff
@@ -33,24 +33,24 @@
 import java.util.function.Function;
 
 /**
- * An operation in general form that is utilized to construct a concrete subclass of {@link Op},
- * such as a subclass of {@link OpWithDefinition}, associated with the operation name.
+ * An operation's external content (a record) that can be utilized to construct an instance
+ * of an {@link ExternalizableOp} associated with the operation's name.
  *
  * @param name            the operation name
  * @param operands        the list of operands
  * @param successors      the list of successors
  * @param resultType      the operation result type
- * @param attributes      the attribute map, modifiable
- * @param bodyDefinitions the list of body builders for building the bodies
+ * @param attributes      the operation's specific content as an attributes map, modifiable
+ * @param bodyDefinitions the list of body builders for building the operation's bodies
  * @apiNote Deserializers of operations may utilize this record to construct operations,
  * thereby separating the specifics of deserializing from construction.
  */
-public record OpDefinition(String name,
-                           List<Value> operands,
-                           List<Block.Reference> successors,
-                           TypeElement resultType,
-                           Map<String, Object> attributes,
-                           List<Body.Builder> bodyDefinitions) {
+public record ExternalOpContent(String name,
```

Comment:
Wouldn't `ExternalizedOp` or `SerializedOp` be better - e.g. more akin to `SerializableLambda` ?

Reply:
I nearly chose `ExternalizedOp` but thought it looked to similar to `ExternalizableOp`. But, most of the time one operates the former. I will rename to `ExternalizedOp` and also enclose it within `ExternalizableOp`.

Leaning into this, in a following PR I will add abstract externalizable op types classes for grouping the core and extended ops. 

We can probably apply the same patterns to `TypeElement` and `TypeDefinition` in a following PR.

Reply:
Updated PR as described above.